azurerm/internal/services/apimanagement/api_management_resource.go

```diff
@@ -899,17 +914,38 @@ func flattenApiManagementAdditionalLocations(input *[]apimanagement.AdditionalLo
 	return results
 }
 
-func expandAzureRmApiManagementIdentity(d *schema.ResourceData) *apimanagement.ServiceIdentity {
+func expandAzureRmApiManagementIdentity(d *schema.ResourceData) (*apimanagement.ServiceIdentity, error) {
 	vs := d.Get("identity").([]interface{})
 	if len(vs) == 0 {
-		return nil
+		return &apimanagement.ServiceIdentity{
+			Type: apimanagement.None,
+		}, nil
 	}
 
 	v := vs[0].(map[string]interface{})
 	identityType := v["type"].(string)
-	return &apimanagement.ServiceIdentity{
+
+	managedServiceIdentity := apimanagement.ServiceIdentity{
 		Type: apimanagement.ApimIdentityType(identityType),
 	}
+
+	identityIdSet := (v["identity_ids"].(*schema.Set))
+	if managedServiceIdentity.Type == apimanagement.UserAssigned || managedServiceIdentity.Type == apimanagement.SystemAssignedUserAssigned {
+		if identityIdSet.Len() == 0 {
+			return nil, fmt.Errorf("`identity_ids` must have at least 1 element when `type` includes `UserAssigned`")
+		}
```

Comment:
Since it isn't `Required`, when `type` contains `UserAssigned` and `identity_ids` is not specified, I get an empty set. That was at least my experience.

Reply:
Oh, you are right. The user might not assign this attribute, then we will get an empty set.